go/ephemeral/device_ek.go

```diff
@@ -45,99 +48,101 @@ func postNewDeviceEK(ctx context.Context, g *libkb.GlobalContext, sig string) er
 }
 
 func getServerMaxDeviceEK(ctx context.Context, g *libkb.GlobalContext) (maxGeneration keybase1.EkGeneration, err error) {
-	deviceEKs, err := GetOwnActiveDeviceEKMetadata(ctx, g)
+	deviceEKs, err := GetActiveDeviceEKMetadata(ctx, g)
 	if err != nil {
 		return maxGeneration, err
 	}
-
-	for _, deviceEK := range deviceEKs {
-		if deviceEK.Generation > maxGeneration {
-			maxGeneration = deviceEK.Generation
-		}
+	deviceID := g.Env.GetDeviceID()
+	metadata, ok := deviceEKs[deviceID]
+	if ok {
+		return metadata.Generation, nil
 	}
+	// We may not have an EK yet, let's try with this and fail if the server
+	// rejects.
+	g.Log.CDebugf(ctx, "No deviceEK found on the server")
 	return maxGeneration, nil
```

Comment:
It seems like the intention is to return zero if the server gave us nothing? Let's add an explicit comment about that.

Reply:
just changed to return `0`, self documenting ;)